src/r2b2/audit.py

```diff
@@ -249,56 +272,52 @@ def run(self):
         previous_votes_for_winner = 0
+        curr_round = 0
 
         while sample_size < max_sample_size:
+            curr_round += 1
+            print('\n----------\n{:^10}\n----------\n'.format('Round {}'.format(curr_round)))
+
+            if verbose and curr_round > 1:
+                click.echo('\n+--------------------------------------------------+')
+                click.echo('|{:^50}|'.format('Audit Statistics'))
+                click.echo('|{:50}|'.format(' '))
+                click.echo('|{:<50}|'.format('Minimum Sample Size: {}'.format(self.min_sample_size)))
+                click.echo('|{:<50}|'.format('Maximum Sample Size: {}'.format(max_sample_size)))
+                click.echo('|{:50}|'.format(' '))
+                click.echo('|{:^16}|{:^16}|{:^16}|'.format('Round', 'Stopping Prob.', 'Risk'))
+                click.echo('|----------------|----------------|----------------|')
+                for r in range(1, curr_round):
+                    click.echo('|{:^16}|{:^16}|{:^16}|'.format(r, '{:.12f}'.format(self.stopping_prob_schedule[r - 1]),
+                                                               '{:.12f}'.format(self.risk_schedule[r - 1])))
+                click.echo('+--------------------------------------------------+')
+
             self.next_sample_size()
 
-            while True:
-                sample_size = int(input('Enter next sample size (as a running total): '))
-                if sample_size < 1:
-                    print('Invalid Input: Sample size must be greater than 1.')
-                    continue
-                if len(self.rounds) > 0 and sample_size <= self.rounds[-1]:
-                    print('Invalid Input: Sample size must be greater than previous round.')
-                    continue
-                if sample_size < self.min_sample_size:
-                    print('Invalid Input: Sample size must be larger than ', self.min_sample_size)
-                    continue
-                if sample_size > max_sample_size:
-                    print('Invalid Input: Sample size cannot exceed ', max_sample_size)
-                    continue
-                break
+            prev_sample_size = sample_size
+            sample_size = click.prompt('Enter next sample size (as a running total)',
+                                       type=click.IntRange(prev_sample_size + 1, max_sample_size))
+            self.rounds.append(sample_size)
 
-            while sample_size < 1 or sample_size > max_sample_size:
-                print('Invalid sample size! Please enter a sample size between 1 and ', max_sample_size)
-                sample_size = int(input('Enter next sample size (as a running total): '))
+            votes_for_winner = click.prompt('Enter total number of votes for reported winner found in sample',
+                                            type=click.IntRange(previous_votes_for_winner, sample_size))
```

Comment:
Could be made a tad more precise: IntRange(previous votes for winner, previous votes for winner + (this round's sample size - previous round's sample size).

Reply:
This value is also a running total, so the max would be the full sample size. I should probably make the wording of the prompts better. Or we could change the execution to take per-round values and total them in the code.

Reply:
Maybe I'm misunderstanding, and please correct me if I am, but if we had 2 rounds of 200 and 400 (cumulative), and you input that you got, say, 100 for winner on the first round, then the possibilities for winner ballots in the second round are [100, 300], no?

Reply:
Yup. I misread the above as `sample_size - previous_sample_size` instead of `previous_votes_for_winner + (sample_size - previous_sample_size)`, my bad.